brave/brave-core
# Add HTTPS-Only/First metric

Resolves https://github.com/brave/brave-browser/issues/35819



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start with a fresh profile, wait 30 seconds, ensure metric has value 0.
1. Start loading some normal HTTPS websites. Restart browser, wait 30 seconds, ensure metric is still 0.
2. Try loading one or more HTTP-only sites (i.e. http://httpforever.com). Restart browser, wait 30 seconds, ensure metric corresponds to the expected percentage (i.e. If you visited 2 HTTPS sites, and one HTTP site, the percentage would be 33%)
3. Reload the same HTTP-only sites that were previously visited. Restart browser, wait 30 seconds, ensure metric does not change. (Automatically allowlisted HTTP sites should not count towards the percentage).
4. Load some more HTTPS websites, ensure the metric changes according to the expected percentage after browser restarts and waiting 30 seconds.
